go/worker/keymanager/watcher.go

```diff
@@ -321,6 +352,14 @@ func (w *rtNodeWatcher) watch(ctx context.Context) {
 				// nodes have been set (even if the new set is empty).
 				continue
 			}
+		case ne := <-nodeCh:
+			if ne.IsRegistration {
+				if ne.Node.HasRoles(node.RoleObserver) && ne.Node.HasRuntime(w.runtimeID) {
+					allowableNodes[ne.Node.ID] = struct{}{}
+				}
```

Comment:
Node should probably be deleted in case the roles/runtimes change in a way that the above condition no longer holds.